pkl/evaluator_manager_exec_unix.go

```diff
@@ -0,0 +1,34 @@
+//go:build !windows
+// +build !windows
+
+// ===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ===----------------------------------------------------------------------===//
+package pkl
+
+import (
+	"os"
+	"syscall"
+)
+
+// killProcess kills the process's entire group
+func killProcess(proc *os.Process) error {
+	pgid, err := syscall.Getpgid(proc.Pid)
```

Comment:
For this to work, I believe we need to set `cmd.SysProcAttr`.

See https://stackoverflow.com/a/29552044

Note that `SysProcAttr.Setpgid` isn't available on Windows, so we need to place the cmd init logic inside these platform-specific files too.